packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -485,8 +485,8 @@ class _CheckboxState extends State<Checkbox> with TickerProviderStateMixin, Togg
           ..focusColor = effectiveFocusOverlayColor
           ..splashRadius = widget.splashRadius ?? checkboxTheme.splashRadius ?? kRadialReactionRadius
           ..downPosition = downPosition
-          ..isFocused = states.contains(MaterialState.focused)
-          ..isHovered = states.contains(MaterialState.hovered)
+          ..isFocused = states.isFocused
+          ..isHovered = states.isHovered
```

Comment:
I've applied the edits implied by this PR's core changes to just one file, `checkbox.dart`, chosen pseudo-randomly, as a demonstration. If the idea is accepted, I will apply these changes across the framework.